ksql-engine/src/main/java/io/confluent/ksql/topic/TopicDeleteInjector.java

```diff
@@ -99,22 +101,21 @@ public TopicDeleteInjector(
             () -> topicClient.deleteTopics(ImmutableList.of(source.getKafkaTopicName())),
             ExecutorUtil.RetryBehaviour.ALWAYS);
       } catch (Exception e) {
-        throw new KsqlException("Could not delete the corresponding kafka topic: "
-            + source.getKafkaTopicName(), e);
+        throw new RuntimeException("Could not delete the corresponding kafka topic: "
+                + sourceName, e);
       }
 
-      if (source.getValueSerdeFactory().getFormat() == Format.AVRO) {
-        try {
+      try {
+        if (source.getValueSerdeFactory().getFormat() == Format.AVRO) {
```

Comment:
not that I'm opposed, but why did we decide to move the if block inside the try?

Reply:
I think it looks a bit cleaner with two try blocks at the same level. But honestly I'm not sure how this ended up happening.